File: app/src/main/java/com/osrsd/command/PrintItems.java
//package com.osrsd.command;
//
//import com.osrsd.App;
//import com.osrsd.cache.Library;
//import com.osrsd.cache.loader.ItemLoader;
//import com.osrsd.spring.domain.ItemDefinition;
//import lombok.AllArgsConstructor;
//
//import java.util.concurrent.CountDownLatch;
//
//@AllArgsConstructor
//public class PrintItems implements Runnable {
//
//    private final CountDownLatch latch;
//
//    @Override
//    public void run() {
//        long start = System.currentTimeMillis();
//        Library.cachePrintable(ItemDefinition.class, new ItemLoader().load().getDefinitions());
//        App.prompt(PrintItems.class, start);
//        latch.countDown();
//    }
//
//}

File: app/src/main/java/com/osrsd/command/PrintItems.kt
package com.osrsd.command

import com.osrsd.cache.Library
import com.osrsd.cache.loader.ItemLoader
import com.osrsd.spring.App
import com.osrsd.spring.domain.ItemDefinition
import java.util.concurrent.CountDownLatch

class PrintItems(val latch: CountDownLatch) : Runnable {

    override fun run() {
        val start = System.currentTimeMillis()
        Library.store(ItemDefinition::class.java, ItemLoader().load().definitions)
        App.prompt(PrintItems::class.java, start)
        latch.countDown()
    }

}